scripts/build_images/build.py

```diff
@@ -0,0 +1,205 @@
+
+
+platform = get_current_platform()
+
+built_images: set[str] = set()
+
+with local_registry(REGISTRY_NAME) if has_base_image else nullcontext():
+    for image in image_list:
+        is_base_image = image in BASE_IMAGES
+        base_image = BASE_IMAGE_MAPPING.get(image)
+        base_image_built = base_image in built_images
+        image_path = get_image_path(image)
+        was_modified = check_path_modified(image_path)
+
+        if not was_modified and not base_image_built:
+            print(f"Skipping {image} because it hasn't changed.")
+            continue
+
+        built_images.add(image)
+
+        # Make temporary files for the metadata.
+        with tempfile.NamedTemporaryFile(delete=False) as metadata_file:
+            pass
+
+        args = [
+            "docker",
+            "buildx",
+            "build",
+            "--platform",
+            platform,
+            "--no-cache",
+            "--tag",
+            f"localhost:5000/{image}",
+            "--progress",
+            "plain",
+            "--metadata-file",
+            metadata_file.name,
+            # "--load",
+            "--output=type=image,push-by-digest=true,name-canonical=true,push=true",
+        ]
+
+        if base_image and base_image_built:
+            args.extend([
+                # We have some images that rely on other images. They're configured to
+                # use this arg to determine which base image tag to use.
+                "--build-arg",
+                f"BASE_IMAGE_TAG={tag}",
+                "--build-arg",
+                "BASE_IMAGE_REGISTRY=localhost:5000",
+            ])
+
+        if should_push:
+            args.extend([
+                # Only tag it with the registry name if we're going to push.
+                "--tag",
+                image,
+            ])
+
+        args.extend([image_path])
+
+        print(f"Building image {image} for platform {platform}")
+        print_and_run_command(args)
+
+        with open(metadata_file.name) as f:
+            metadata = f.read()
+
+        print(f"Metadata: {metadata.strip()}")
+        digest = json.loads(metadata)["containerimage.digest"]
+
+        if is_base_image:
+            # `buildx` cannot use locally-built images when the
+            # `docker-container` driver is used, and that driver must be used in
+            # order to support `push-by-digest`. To make the base image available,
+            # we'll push it to a local registry.
+            #
+            # https://github.com/moby/buildkit/issues/2343
+            print_and_run_command([
+                "docker",
+                "pull",
+                f"localhost:5000/{image}@{digest}",
+            ])
+            print_and_run_command([
+                "docker",
+                "tag",
+                f"localhost:5000/{image}@{digest}",
+                f"localhost:5000/{image}:{tag}",
+            ])
+            print_and_run_command([
```

Comment:
In CI, this docker push step will take an extra 3 minutes per base image (!).

Reply:
Well that is much  faster than what I had observed in CI on my branch. I wonder if that is due to the tmpfs mount. Anything <1min feels like acceptable overhead.